lib/puma/server.rb

```diff
@@ -344,12 +343,33 @@ def handle_check
       return false
     end
 
+    # Runs the provided block if the Reactor queue is enabled.
+    # Returns block return value if queue is enabled, or nil if queue is not enabled.
+    def if_queue_requests
+      @queue_mutex.synchronize do
+        if @queue_requests
```

Comment:
Since queue_requests is set at configure-time, I think we can move this outside the mutex, right?